storage_service/locations/models/pipeline_local.py

```diff
@@ -136,3 +136,17 @@ def move_from_storage_service(self, source_path, destination_path, package=None)
 
         # Move file
         return self.space.move_rsync(source_path, destination_path, assume_rsync_daemon=self.assume_rsync_daemon, rsync_password=self.rsync_password)
+
+    def isfile(self, path_):
+        """Verify that something is a file in this Space's context."""
+        LOGGER.info("Testing if isfile in pipeline file system: %s", path_)
+        basename = os.path.basename(path_)
+        if not len(basename):
+            return False
+        path_ = os.path.join(os.path.dirname(path_), "")
+        ssh_path = self._format_host_path(path_)
+        location_entries = self.space.browse_rsync(ssh_path, assume_rsync_daemon=self.assume_rsync_daemon, rsync_password=self.rsync_password)
+        if basename in location_entries.get("entries", []) \
+                and basename not in location_entries.get("directories", []):
```

Comment:
Doesn't have a `files` key? Just making sure, it could be a great addition in the future. I also prefer the parenthesis syntax for multi-line ifs, I guess black will help us with that ;)

Reply:
Also, you could return the condition directly and remove the two returns bellow.

Reply:
:+1:

Reply:
Unfortunately no, there is a `properties` so one could check if it has 'size'. That being said, I don't know if you read this the same as me, but we're not actually attaching the `properties dict` to anything so it's not returned for this function??? (My eyes might just be tired)

https://github.com/artefactual/archivematica-storage-service/blob/d83d383f44dd258f6ba4c1b5a879ca22270e2ae6/storage_service/locations/models/space.py#L653-L668

Reply:
What I mean is that you could use:

```

Reply:
Yup :+1: that's in my changes :nerd_face:

Reply:
Right, I misunderstood the comment. Yes, not for now, but checking and returning an extra key for files in `browse_rsync` could be a great addition in the future.